packages/block-library/src/group/variations.js

```diff
@@ -4,12 +4,21 @@
 import { __ } from '@wordpress/i18n';
 
 const variations = [
+	{
+		name: 'group',
+		title: __( 'Group' ),
+		description: __( 'Blocks shown in a column.' ),
+		attributes: { layout: { type: 'default' } },
+		scope: [ 'transform' ],
+		isActive: ( blockAttributes ) =>
+			blockAttributes.layout?.type === 'default',
```

Comment:
When we insert a group block, it's not checked because an undefined layout is also a "group" block. So we should add `! blockAttributes.layout || blockAttributes.layout?.type === 'default'` here